### ISSUE TITLE
negative loss when training PPOAgent

### ISSUE THREAD

**Issue by marooncn** [tensorforce/tensorforce#567]
Hi, When I train a PPOAgent:
```
network_spec = [
    dict(type='dense', size=512, activation='relu'),
    dict(type='dense', size=512, activation='relu'),
    dict(type='dense', size=512, activation='relu'),
]

optimizer = dict(
    type='adam',
    learning_rate=0.0001
)

agent = PPOAgent(
    states=dict(shape=(36,), type='float'), 
    actions=dict(linear_vel=dict(shape=(), type='float', min_value=0.0, max_value=1.0),
                 angular_vel=dict(shape=(), type='float', min_value=-1.0, max_value=1.0)),
    network=network_spec,
    saver=dict(directory='./models', basename='PPO_model.ckpt', load=restore),
    summarizer=dict(directory='./record/PPO', labels=["graph", "losses", "reward"]),
    step_optimizer=optimizer
)
```
I find that the loss is always negative which shouldn't happen:
![default](https://user-images.githubusercontent.com/29606478/53707214-f9cd1a00-3e68-11e9-9f7d-7829403994de.png)
By the way, only "losses" is  recorded in tensorboard, why can't "reward" be recorded?

**Comment by AlexKuhnle** [comment]
Hey @marooncn,

First, regarding the negative loss, that's actually expected behavior. Why? Policy gradient methods do not *minimize* an objective (like squared difference of Q-values, as in DQN), but *maximize* the expected return, which translates to a minimization of the objective value in the implementation (since TensorFlow's and Tensorforce's optimizers always *minimize* a target). This is untypical in a supervised context, where losses are usually distances and hence positive, but it's not a problem in principle.

Regarding TensorBoard, unfortunately the current architecture as is doesn't work well with most summaries. This is one of the motivations behind the [major-revision branch](https://github.com/tensorforce/tensorforce/tree/major-revision) that I'm currently working on, which will hopefully soon become the new master version. I encourage you to check it out, as e.g. summaries work much better there. Documentation is not great so far, but I'm happy to help with the move, if you want to ping me on e.g. Gitter.

**Comment by marooncn** [comment]
@AlexKuhnle Thanks for your response, I'm still confused about the negative loss. How to evaluate the performance in this case? Which one is better, loss is -0.1 or -0.2?

**Comment by AlexKuhnle** [comment]
The loss is always minimized in Tensorforce, so -0.2 is better than -0.1. Think of it as roughly reflecting the negative expected return.

Side note: Best not to rely only on the loss for evaluation, since it probably doesn't reflect actual performance improvements too well. Better to only look at it to see whether *something is happening*.

**Comment by marooncn** [comment]
@AlexKuhnle Thank you and actually that's why I want to record the reward in Tensorboard. I will try [major-revision branch](https://github.com/tensorforce/tensorforce/tree/major-revision).
